lua/FTerm/terminal.lua

```diff
@@ -150,16 +150,23 @@ end
 
 -- Terminal:open does all the magic of opening terminal
 function Terminal:open()
-    self:remember_cursor()
 
-    local buf = self:create_buf()
+    -- Create new window and terminal if it doesn't exist
+    if not vim.tbl_contains(vim.api.nvim_list_wins(), self.win) then
```

Comment:
I think a `nvim_win_is_valid(self.win)` check would also suffice. thoughts?

Reply:
In lua not having an argument at all will be interpreted as nil, which means that `nvim_win_is_valid()` is the same as `nvim_win_is_valid(nil)`, which checks current window if it is valid (which it usually is). Since initially `self.win=nil` and we want it to be interpreted as invalid, we can not use `nvim_win_is_valid` as you propose. But it is a good thought.

Reply:
TIL about the `nil` argument behaviour 😂. We can also check for `nil` before using `nvim_win_is_valid`(?).

Reply:
Perhaps I could suggest that we instead considered to change the default values in `Terminal:new()`? Like `win=-1, buf=-1, terminal=-1, jobid=-1`? These values are invalid ids, but unlike nil will often raise errors when used internally in the nvim api. It will generally reduce the likelihood of bugs related to uninitiated variables and `nvim_win_is_valid` can be used as-is wihtout an extra clause.

Reply:
And just to note, the -1 value is used like this internally in the api as well. Like `jobstart(..)` will return -1 for invalid executable, and `bufnr(..) = -1` and `winnr(..) = -1` indicate buffer and window does not exist respectivly.